server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java

```diff
@@ -213,63 +214,57 @@ public void testNoMasterActionsWriteMasterBlock() throws Exception {
-                    internalCluster().getNodeNames()).filter(n -> n.equals(nodeName) == false).findAny().get());
-
-                assertTrue(awaitBusy(() -> {
-                        ClusterState state = remainingClient.admin().cluster().prepareState().setLocal(true).get().getState();
-                        return state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID);
-                    }
-                ));
-
-                GetResponse getResponse = remainingClient.prepareGet("test1", "type1", "1").get();
-                assertExists(getResponse);
-
-                SearchResponse countResponse = remainingClient.prepareSearch("test1").setAllowPartialSearchResults(true).setSize(0).get();
-                assertHitCount(countResponse, 1L);
-
-                logger.info("--> here 3");
-                SearchResponse searchResponse = remainingClient.prepareSearch("test1").setAllowPartialSearchResults(true).get();
-                assertHitCount(searchResponse, 1L);
-
-                countResponse = remainingClient.prepareSearch("test2").setAllowPartialSearchResults(true).setSize(0).get();
-                assertThat(countResponse.getTotalShards(), equalTo(3));
-                assertThat(countResponse.getSuccessfulShards(), equalTo(1));
-
-                TimeValue timeout = TimeValue.timeValueMillis(200);
-                long now = System.currentTimeMillis();
-                try {
-                    remainingClient.prepareUpdate("test1", "type1", "1")
-                        .setDoc(Requests.INDEX_CONTENT_TYPE, "field", "value2").setTimeout(timeout).get();
-                    fail("Expected ClusterBlockException");
-                } catch (ClusterBlockException e) {
-                    assertThat(System.currentTimeMillis() - now, greaterThan(timeout.millis() - 50));
-                    assertThat(e.status(), equalTo(RestStatus.SERVICE_UNAVAILABLE));
-                } catch (Exception e) {
-                    logger.info("unexpected", e);
-                    throw e;
-                }
-
-                try {
-                    remainingClient.prepareIndex("test1", "type1", "1")
-                        .setSource(XContentFactory.jsonBuilder().startObject().endObject()).setTimeout(timeout).get();
-                    fail("Expected ClusterBlockException");
-                } catch (ClusterBlockException e) {
-                    assertThat(e.status(), equalTo(RestStatus.SERVICE_UNAVAILABLE));
-                }
-
-                logger.info("finished assertions, restarting node [{}]", nodeName);
-
-                return Settings.EMPTY;
+        final NetworkDisruption disruptionScheme
+            = new NetworkDisruption(new IsolateAllNodes(new HashSet<>(nodes)), new NetworkDisconnect());
+        internalCluster().setDisruptionScheme(disruptionScheme);
+        disruptionScheme.startDisrupting();
+
+        final Client remainingClient = client();
```

Comment:
I'm confused. Why capture the client here and not just use `client()` at usage sites below?

Reply:
We first check that we have a client to a node that has applied the no-master block. The alternative would be to wait for _all_ nodes to apply the no-master block before proceeding.

Reply:
However the name was a hangover from the earlier implementation that shut nodes down, so I renamed it in 788df4a27ebe265bf28da9764b14ea3a222e0c82.